Cmake and "No rule to make target 'static'.  Stop"

I'm attempting to build the following [standard targets](http://cryptopp.com/wiki/GNUmakefile#Standard_Targets): `static`, `dynamic`, and `test`.:

```
$ cmake .
-- The C compiler identification is AppleClang 5.1.0.5030040
-- The CXX compiler identification is AppleClang 5.1.0.5030040
-- Check for working C compiler: /Applications/Xcode-5_1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Check for working C compiler: /Applications/Xcode-5_1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Applications/Xcode-5_1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
-- Check for working CXX compiler: /Applications/Xcode-5_1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Check if the system is big endian
-- Searching 16 bit integer
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of unsigned short
-- Check size of unsigned short - done
-- Using unsigned short
-- Check if the system is big endian - little endian
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - found
-- Found Threads: TRUE  
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/jwalton/cryptopp
```

Then:
```
$ make -f Makefile static dynamic test
make: *** No rule to make target 'static'.  Stop.
```

And:
```
$ uname -a
Darwin riemann.local 12.6.0 Darwin Kernel Version 12.6.0: Wed Mar 18 16:23:48 PDT 2015; root:xnu-2050.48.19~1/RELEASE_X86_64 x86_64
```

It seems to be a problem with environment. On my Mac OS X 10.11.5 running Xcode-7.3.1 and CMake-3.5.2 installed by Macports, I'm getting:
```
$ mkdir build
$ cd build
$ cmake ..
-- The C compiler identification is AppleClang 7.3.0.7030031
-- The CXX compiler identification is AppleClang 7.3.0.7030031
-- Check for working C compiler: /opt/local/bin/clang
-- Check for working C compiler: /opt/local/bin/clang -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /opt/local/bin/clang++
-- Check for working CXX compiler: /opt/local/bin/clang++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Check if the system is big endian
-- Searching 16 bit integer
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of unsigned short
-- Check size of unsigned short - done
-- Using unsigned short
-- Check if the system is big endian - little endian
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - found
-- Found Threads: TRUE
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/uri/src/cryptopp/build
$ make
Scanning dependencies of target cryptopp-object
[  1%] Building CXX object CMakeFiles/cryptopp-object.dir/cryptlib.cpp.o
[  1%] Building CXX object CMakeFiles/cryptopp-object.dir/cpu.cpp.o
[  2%] Building CXX object CMakeFiles/cryptopp-object.dir/3way.cpp.o
[  3%] Building CXX object CMakeFiles/cryptopp-object.dir/adler32.cpp.o
[  4%] Building CXX object CMakeFiles/cryptopp-object.dir/algebra.cpp.o
[  4%] Building CXX object CMakeFiles/cryptopp-object.dir/algparam.cpp.o
[  5%] Building CXX object CMakeFiles/cryptopp-object.dir/arc4.cpp.o
[  6%] Building CXX object CMakeFiles/cryptopp-object.dir/asn.cpp.o
[  7%] Building CXX object CMakeFiles/cryptopp-object.dir/authenc.cpp.o
[  7%] Building CXX object CMakeFiles/cryptopp-object.dir/base32.cpp.o
[  8%] Building CXX object CMakeFiles/cryptopp-object.dir/base64.cpp.o
[  9%] Building CXX object CMakeFiles/cryptopp-object.dir/basecode.cpp.o
[  9%] Building CXX object CMakeFiles/cryptopp-object.dir/bfinit.cpp.o
[ 10%] Building CXX object CMakeFiles/cryptopp-object.dir/blake2.cpp.o
[ 11%] Building CXX object CMakeFiles/cryptopp-object.dir/blowfish.cpp.o
[ 12%] Building CXX object CMakeFiles/cryptopp-object.dir/blumshub.cpp.o
[ 12%] Building CXX object CMakeFiles/cryptopp-object.dir/camellia.cpp.o
[ 13%] Building CXX object CMakeFiles/cryptopp-object.dir/cast.cpp.o
[ 14%] Building CXX object CMakeFiles/cryptopp-object.dir/casts.cpp.o
[ 15%] Building CXX object CMakeFiles/cryptopp-object.dir/cbcmac.cpp.o
[ 15%] Building CXX object CMakeFiles/cryptopp-object.dir/ccm.cpp.o
[ 16%] Building CXX object CMakeFiles/cryptopp-object.dir/chacha.cpp.o
[ 17%] Building CXX object CMakeFiles/cryptopp-object.dir/channels.cpp.o
[ 18%] Building CXX object CMakeFiles/cryptopp-object.dir/cmac.cpp.o
[ 18%] Building CXX object CMakeFiles/cryptopp-object.dir/crc.cpp.o
[ 19%] Building CXX object CMakeFiles/cryptopp-object.dir/default.cpp.o
In file included from /Users/uri/src/cryptopp/default.cpp:13:
/Users/uri/src/cryptopp/default.h:164:43: warning: 'DefaultDecryptor' is deprecated
      [-Wdeprecated-declarations]
        class MACBadErr : public DefaultDecryptor::Err {public: MACBadErr() ...
                                                 ^
/Users/uri/src/cryptopp/default.h:63:7: note: 'DefaultDecryptor' has been explicitly
      marked deprecated here
class DefaultDecryptor : public ProxyFilter
      ^
/Users/uri/src/cryptopp/default.h:179:18: warning: 'DefaultDecryptor' is deprecated
      [-Wdeprecated-declarations]
        DefaultDecryptor::State CurrentState() const;
                        ^
/Users/uri/src/cryptopp/default.h:63:7: note: 'DefaultDecryptor' has been explicitly
      marked deprecated here
class DefaultDecryptor : public ProxyFilter
      ^
2 warnings generated.
[ 20%] Building CXX object CMakeFiles/cryptopp-object.dir/des.cpp.o
[ 20%] Building CXX object CMakeFiles/cryptopp-object.dir/dessp.cpp.o
[ 21%] Building CXX object CMakeFiles/cryptopp-object.dir/dh.cpp.o
[ 22%] Building CXX object CMakeFiles/cryptopp-object.dir/dh2.cpp.o
[ 23%] Building CXX object CMakeFiles/cryptopp-object.dir/dll.cpp.o
[ 23%] Building CXX object CMakeFiles/cryptopp-object.dir/dsa.cpp.o
[ 24%] Building CXX object CMakeFiles/cryptopp-object.dir/eax.cpp.o
[ 25%] Building CXX object CMakeFiles/cryptopp-object.dir/ec2n.cpp.o
[ 26%] Building CXX object CMakeFiles/cryptopp-object.dir/eccrypto.cpp.o
[ 26%] Building CXX object CMakeFiles/cryptopp-object.dir/ecp.cpp.o
[ 27%] Building CXX object CMakeFiles/cryptopp-object.dir/elgamal.cpp.o
[ 28%] Building CXX object CMakeFiles/cryptopp-object.dir/emsa2.cpp.o
[ 28%] Building CXX object CMakeFiles/cryptopp-object.dir/eprecomp.cpp.o
[ 29%] Building CXX object CMakeFiles/cryptopp-object.dir/esign.cpp.o
[ 30%] Building CXX object CMakeFiles/cryptopp-object.dir/fhmqv.cpp.o
[ 31%] Building CXX object CMakeFiles/cryptopp-object.dir/files.cpp.o
[ 31%] Building CXX object CMakeFiles/cryptopp-object.dir/filters.cpp.o
[ 32%] Building CXX object CMakeFiles/cryptopp-object.dir/fips140.cpp.o
[ 33%] Building CXX object CMakeFiles/cryptopp-object.dir/gcm.cpp.o
[ 34%] Building CXX object CMakeFiles/cryptopp-object.dir/gf256.cpp.o
[ 34%] Building CXX object CMakeFiles/cryptopp-object.dir/gf2_32.cpp.o
[ 35%] Building CXX object CMakeFiles/cryptopp-object.dir/gf2n.cpp.o
[ 36%] Building CXX object CMakeFiles/cryptopp-object.dir/gfpcrypt.cpp.o
[ 36%] Building CXX object CMakeFiles/cryptopp-object.dir/gost.cpp.o
[ 37%] Building CXX object CMakeFiles/cryptopp-object.dir/gzip.cpp.o
[ 38%] Building CXX object CMakeFiles/cryptopp-object.dir/hex.cpp.o
[ 39%] Building CXX object CMakeFiles/cryptopp-object.dir/hmac.cpp.o
[ 39%] Building CXX object CMakeFiles/cryptopp-object.dir/hrtimer.cpp.o
[ 40%] Building CXX object CMakeFiles/cryptopp-object.dir/ida.cpp.o
[ 41%] Building CXX object CMakeFiles/cryptopp-object.dir/idea.cpp.o
[ 42%] Building CXX object CMakeFiles/cryptopp-object.dir/integer.cpp.o
[ 42%] Building CXX object CMakeFiles/cryptopp-object.dir/iterhash.cpp.o
[ 43%] Building CXX object CMakeFiles/cryptopp-object.dir/luc.cpp.o
[ 44%] Building CXX object CMakeFiles/cryptopp-object.dir/mars.cpp.o
[ 45%] Building CXX object CMakeFiles/cryptopp-object.dir/marss.cpp.o
[ 45%] Building CXX object CMakeFiles/cryptopp-object.dir/md2.cpp.o
[ 46%] Building CXX object CMakeFiles/cryptopp-object.dir/md4.cpp.o
[ 47%] Building CXX object CMakeFiles/cryptopp-object.dir/md5.cpp.o
[ 47%] Building CXX object CMakeFiles/cryptopp-object.dir/misc.cpp.o
[ 48%] Building CXX object CMakeFiles/cryptopp-object.dir/modes.cpp.o
[ 49%] Building CXX object CMakeFiles/cryptopp-object.dir/mqueue.cpp.o
[ 50%] Building CXX object CMakeFiles/cryptopp-object.dir/mqv.cpp.o
[ 50%] Building CXX object CMakeFiles/cryptopp-object.dir/nbtheory.cpp.o
[ 51%] Building CXX object CMakeFiles/cryptopp-object.dir/network.cpp.o
[ 52%] Building CXX object CMakeFiles/cryptopp-object.dir/oaep.cpp.o
[ 53%] Building CXX object CMakeFiles/cryptopp-object.dir/osrng.cpp.o
[ 53%] Building CXX object CMakeFiles/cryptopp-object.dir/panama.cpp.o
[ 54%] Building CXX object CMakeFiles/cryptopp-object.dir/pkcspad.cpp.o
[ 55%] Building CXX object CMakeFiles/cryptopp-object.dir/polynomi.cpp.o
[ 55%] Building CXX object CMakeFiles/cryptopp-object.dir/pssr.cpp.o
[ 56%] Building CXX object CMakeFiles/cryptopp-object.dir/pubkey.cpp.o
[ 57%] Building CXX object CMakeFiles/cryptopp-object.dir/queue.cpp.o
[ 58%] Building CXX object CMakeFiles/cryptopp-object.dir/rabin.cpp.o
[ 58%] Building CXX object CMakeFiles/cryptopp-object.dir/randpool.cpp.o
[ 59%] Building CXX object CMakeFiles/cryptopp-object.dir/rc2.cpp.o
[ 60%] Building CXX object CMakeFiles/cryptopp-object.dir/rc5.cpp.o
[ 61%] Building CXX object CMakeFiles/cryptopp-object.dir/rc6.cpp.o
[ 61%] Building CXX object CMakeFiles/cryptopp-object.dir/rdrand.cpp.o
[ 62%] Building CXX object CMakeFiles/cryptopp-object.dir/rdtables.cpp.o
[ 63%] Building CXX object CMakeFiles/cryptopp-object.dir/rijndael.cpp.o
[ 63%] Building CXX object CMakeFiles/cryptopp-object.dir/ripemd.cpp.o
[ 64%] Building CXX object CMakeFiles/cryptopp-object.dir/rng.cpp.o
[ 65%] Building CXX object CMakeFiles/cryptopp-object.dir/rsa.cpp.o
[ 66%] Building CXX object CMakeFiles/cryptopp-object.dir/rw.cpp.o
[ 66%] Building CXX object CMakeFiles/cryptopp-object.dir/safer.cpp.o
[ 67%] Building CXX object CMakeFiles/cryptopp-object.dir/salsa.cpp.o
[ 68%] Building CXX object CMakeFiles/cryptopp-object.dir/seal.cpp.o
[ 69%] Building CXX object CMakeFiles/cryptopp-object.dir/seed.cpp.o
[ 69%] Building CXX object CMakeFiles/cryptopp-object.dir/serpent.cpp.o
[ 70%] Building CXX object CMakeFiles/cryptopp-object.dir/sha.cpp.o
[ 71%] Building CXX object CMakeFiles/cryptopp-object.dir/sha3.cpp.o
[ 72%] Building CXX object CMakeFiles/cryptopp-object.dir/shacal2.cpp.o
[ 72%] Building CXX object CMakeFiles/cryptopp-object.dir/shark.cpp.o
[ 73%] Building CXX object CMakeFiles/cryptopp-object.dir/sharkbox.cpp.o
[ 74%] Building CXX object CMakeFiles/cryptopp-object.dir/skipjack.cpp.o
[ 74%] Building CXX object CMakeFiles/cryptopp-object.dir/socketft.cpp.o
[ 75%] Building CXX object CMakeFiles/cryptopp-object.dir/sosemanuk.cpp.o
[ 76%] Building CXX object CMakeFiles/cryptopp-object.dir/square.cpp.o
[ 77%] Building CXX object CMakeFiles/cryptopp-object.dir/squaretb.cpp.o
[ 77%] Building CXX object CMakeFiles/cryptopp-object.dir/strciphr.cpp.o
[ 78%] Building CXX object CMakeFiles/cryptopp-object.dir/tea.cpp.o
[ 79%] Building CXX object CMakeFiles/cryptopp-object.dir/tftables.cpp.o
[ 80%] Building CXX object CMakeFiles/cryptopp-object.dir/tiger.cpp.o
[ 80%] Building CXX object CMakeFiles/cryptopp-object.dir/tigertab.cpp.o
[ 81%] Building CXX object CMakeFiles/cryptopp-object.dir/trdlocal.cpp.o
[ 82%] Building CXX object CMakeFiles/cryptopp-object.dir/ttmac.cpp.o
[ 82%] Building CXX object CMakeFiles/cryptopp-object.dir/twofish.cpp.o
[ 83%] Building CXX object CMakeFiles/cryptopp-object.dir/vmac.cpp.o
[ 84%] Building CXX object CMakeFiles/cryptopp-object.dir/wait.cpp.o
[ 85%] Building CXX object CMakeFiles/cryptopp-object.dir/wake.cpp.o
[ 85%] Building CXX object CMakeFiles/cryptopp-object.dir/whrlpool.cpp.o
[ 86%] Building CXX object CMakeFiles/cryptopp-object.dir/xtr.cpp.o
[ 87%] Building CXX object CMakeFiles/cryptopp-object.dir/xtrcrypt.cpp.o
[ 88%] Building CXX object CMakeFiles/cryptopp-object.dir/zdeflate.cpp.o
[ 88%] Building CXX object CMakeFiles/cryptopp-object.dir/zinflate.cpp.o
[ 89%] Building CXX object CMakeFiles/cryptopp-object.dir/zlib.cpp.o
[ 89%] Built target cryptopp-object
Scanning dependencies of target cryptopp-static
[ 90%] Linking CXX static library libcryptopp.a
/opt/local/bin/ranlib: file: libcryptopp.a(algebra.cpp.o) has no symbols
/opt/local/bin/ranlib: file: libcryptopp.a(eccrypto.cpp.o) has no symbols
/opt/local/bin/ranlib: file: libcryptopp.a(eprecomp.cpp.o) has no symbols
/opt/local/bin/ranlib: file: libcryptopp.a(polynomi.cpp.o) has no symbols
/opt/local/bin/ranlib: file: libcryptopp.a(strciphr.cpp.o) has no symbols
/opt/local/bin/ranlib: file: libcryptopp.a(algebra.cpp.o) has no symbols
/opt/local/bin/ranlib: file: libcryptopp.a(eccrypto.cpp.o) has no symbols
/opt/local/bin/ranlib: file: libcryptopp.a(eprecomp.cpp.o) has no symbols
/opt/local/bin/ranlib: file: libcryptopp.a(polynomi.cpp.o) has no symbols
/opt/local/bin/ranlib: file: libcryptopp.a(strciphr.cpp.o) has no symbols
[ 90%] Built target cryptopp-static
Scanning dependencies of target cryptest-object
[ 91%] Building CXX object CMakeFiles/cryptest-object.dir/bench1.cpp.o
[ 92%] Building CXX object CMakeFiles/cryptest-object.dir/bench2.cpp.o
[ 92%] Building CXX object CMakeFiles/cryptest-object.dir/test.cpp.o
In file included from /Users/uri/src/cryptopp/test.cpp:14:
/Users/uri/src/cryptopp/default.h:164:43: warning: 'DefaultDecryptor' is deprecated
      [-Wdeprecated-declarations]
        class MACBadErr : public DefaultDecryptor::Err {public: MACBadErr() ...
                                                 ^
/Users/uri/src/cryptopp/default.h:63:7: note: 'DefaultDecryptor' has been explicitly
      marked deprecated here
class DefaultDecryptor : public ProxyFilter
      ^
/Users/uri/src/cryptopp/default.h:179:18: warning: 'DefaultDecryptor' is deprecated
      [-Wdeprecated-declarations]
        DefaultDecryptor::State CurrentState() const;
                        ^
/Users/uri/src/cryptopp/default.h:63:7: note: 'DefaultDecryptor' has been explicitly
      marked deprecated here
class DefaultDecryptor : public ProxyFilter
      ^
2 warnings generated.
[ 93%] Building CXX object CMakeFiles/cryptest-object.dir/validat1.cpp.o
[ 94%] Building CXX object CMakeFiles/cryptest-object.dir/validat2.cpp.o
[ 95%] Building CXX object CMakeFiles/cryptest-object.dir/validat3.cpp.o
[ 95%] Building CXX object CMakeFiles/cryptest-object.dir/adhoc.cpp.o
[ 96%] Building CXX object CMakeFiles/cryptest-object.dir/datatest.cpp.o
[ 97%] Building CXX object CMakeFiles/cryptest-object.dir/regtest.cpp.o
[ 98%] Building CXX object CMakeFiles/cryptest-object.dir/fipsalgt.cpp.o
[ 98%] Building CXX object CMakeFiles/cryptest-object.dir/dlltest.cpp.o
[ 99%] Building CXX object CMakeFiles/cryptest-object.dir/fipstest.cpp.o
[ 99%] Built target cryptest-object
Scanning dependencies of target cryptest
[ 99%] Linking CXX executable cryptest
[ 99%] Built target cryptest
Scanning dependencies of target cryptopp-shared
[100%] Linking CXX shared library libcryptopp.dylib
[100%] Built target cryptopp-shared
$ ./cryptest.exe v
. . . . .
All tests passed!

Test ended at Sun Jun 19 03:52:12 2016
Seed used was: 1466322729
$ ll libcryptopp.*
-rwxr-xr-x  1 uri  staff  4286684 Jun 19 03:50 libcryptopp.5.6.dylib*
-rw-r--r--  1 uri  staff  9049192 Jun 19 03:50 libcryptopp.a
lrwxr-xr-x  1 uri  staff       21 Jun 19 03:50 libcryptopp.dylib@ -> libcryptopp.5.6.dylib
$
```

I suspect that `make` gets confused when it finds cmake-generated `Makefile` and `GNUmakefile` in its working directory, but that's just my guess. I can only see that trying to run cmake in the source tree itself seems problematic, while running it in a subdirectory (aka `./build/`) seems to work.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,7 +14,6 @@ set(cryptopp_VERSION_PATCH 3)
 
 include(GNUInstallDirs)
 include(TestBigEndian)
-include(CheckCXXSymbolExists)
 
 #============================================================================
 # Settable options
@@ -124,31 +123,67 @@ endif()
 #============================================================================
 # Compile targets
 #============================================================================
-add_library(cryptopp-object OBJECT ${cryptopp_SOURCES})
 
-if(CMAKE_SIZEOF_VOID_P EQUAL 8)
-	# Enables -fPIC on all 64-bit platforms
-	set_target_properties(cryptopp-object PROPERTIES POSITION_INDEPENDENT_CODE TRUE)
+# Set global includes BEFORE adding any targets for legacy CMake versions
+if(CMAKE_VERSION VERSION_LESS 2.8.12)
+	include_directories("${CMAKE_CURRENT_SOURCE_DIR}")
+endif()
+
+if(NOT CMAKE_VERSION VERSION_LESS 2.8.8)
+	add_library(cryptopp-object OBJECT ${cryptopp_SOURCES})
 endif()
 
 if (BUILD_STATIC)
-	add_library(cryptopp-static STATIC $<TARGET_OBJECTS:cryptopp-object>)
+	if(NOT CMAKE_VERSION VERSION_LESS 2.8.8)
+		add_library(cryptopp-static STATIC $<TARGET_OBJECTS:cryptopp-object>)
+	else()
+		add_library(cryptopp-static STATIC ${cryptopp_SOURCES})
+	endif()
+
 	if (NOT CMAKE_VERSION VERSION_LESS 2.8.12)
 		target_include_directories(cryptopp-static PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}> $<INSTALL_INTERFACE:include/cryptopp>)
-	else()
-		include_directories("${ROOT_SOURCE_DIR}")
 	endif()
 endif()
 
 if (BUILD_SHARED)
-	add_library(cryptopp-shared SHARED $<TARGET_OBJECTS:cryptopp-object>)
+	if(NOT CMAKE_VERSION VERSION_LESS 2.8.8)
+		add_library(cryptopp-shared SHARED $<TARGET_OBJECTS:cryptopp-object>)
+	else()
+		add_library(cryptopp-shared SHARED ${cryptopp_SOURCES})
+	endif()
+
 	if (NOT CMAKE_VERSION VERSION_LESS 2.8.12)
 		target_include_directories(cryptopp-shared PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}> $<INSTALL_INTERFACE:include/cryptopp>)
-	else()
-		include_directories("${ROOT_SOURCE_DIR}")
 	endif()
 endif()
 
+# Set PIC
+if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+	# Enables -fPIC on all 64-bit platforms
+	if(NOT CMAKE_VERSION VERSION_LESS 2.8.9)	# POSITION_INDEPENDENT_CODE support
+		set_target_properties(cryptopp-object PROPERTIES POSITION_INDEPENDENT_CODE TRUE)
+	elseif(NOT CMAKE_VERSION VERSION_LESS 2.8.8)	# Object library support
+		get_target_property(flags_old cryptopp-object COMPILE_FLAGS)
+		set_target_properties(cryptopp-object PROPERTIES COMPILE_FLAGS "${flags_old} -fPIC")
+	else()
+		if (BUILD_STATIC)
+			get_target_property(flags_old_static cryptopp-static COMPILE_FLAGS)
+			if(NOT flags_old_static)
+				set(flags_old_static "")
+			endif()
+			set_target_properties(cryptopp-static PROPERTIES COMPILE_FLAGS "${flags_old_static} -fPIC")
+		endif()
+		if (BUILD_SHARED)
+			get_target_property(flags_old_shared cryptopp-shared COMPILE_FLAGS)
+			if(NOT flags_old_shared)
+				set(flags_old_shared "")
+			endif()
+			set_target_properties(cryptopp-shared PROPERTIES COMPILE_FLAGS "${flags_old_shared} -fPIC")
+		endif()
+	endif()
+endif()
+
+# Set filenames for targets to be "cryptopp"
 if(NOT MSVC)
 	set(COMPAT_VERSION ${cryptopp_VERSION_MAJOR}.${cryptopp_VERSION_MINOR})
 
@@ -163,6 +198,16 @@ if(NOT MSVC)
 				SOVERSION ${COMPAT_VERSION}
 				OUTPUT_NAME cryptopp)
 	endif()
+endif()
+
+# Targets, compatible with Crypto++ GNUMakefile
+if (BUILD_STATIC)
+	add_custom_target(static)
+	add_dependencies(static cryptopp-static)
+endif()
+if (BUILD_SHARED)
+	add_custom_target(dynamic)
+	add_dependencies(dynamic cryptopp-shared)
 endif()
 
 #============================================================================
@@ -190,15 +235,15 @@ endif()
 #============================================================================
 enable_testing()
 if(BUILD_TESTING)
-	add_library(cryptest-object OBJECT ${cryptopp_SOURCES_TEST})
-
-	add_executable(cryptest $<TARGET_OBJECTS:cryptest-object>)
+	add_executable(cryptest ${cryptopp_SOURCES_TEST})
 	target_link_libraries(cryptest cryptopp-static)
 
 	file(COPY ${CMAKE_CURRENT_SOURCE_DIR}/TestData DESTINATION ${PROJECT_BINARY_DIR})
 	file(COPY ${CMAKE_CURRENT_SOURCE_DIR}/TestVectors DESTINATION ${PROJECT_BINARY_DIR})
 
+	add_test(NAME build_cryptest COMMAND "${CMAKE_COMMAND}" --build ${CMAKE_BINARY_DIR} --target cryptest)
 	add_test(NAME cryptest COMMAND $<TARGET_FILE:cryptest> v)
+	set_tests_properties(cryptest PROPERTIES DEPENDS build_cryptest)
 endif()
 
 #============================================================================
@@ -242,10 +287,12 @@ endif()
 install(FILES ${cryptopp_HEADERS} DESTINATION include/cryptopp)
 
 # CMake Package
-include(CMakePackageConfigHelpers)
-write_basic_package_version_file("${PROJECT_BINARY_DIR}/cryptopp-config-version.cmake" VERSION ${cryptopp_VERSION_MAJOR}.${cryptopp_VERSION_MINOR}.${cryptopp_VERSION_PATCH} COMPATIBILITY SameMajorVersion)
-install(FILES cryptopp-config.cmake ${PROJECT_BINARY_DIR}/cryptopp-config-version.cmake DESTINATION "lib/cmake/cryptopp")
-install(EXPORT ${export_name} DESTINATION "lib/cmake/cryptopp")
+if(NOT CMAKE_VERSION VERSION_LESS 2.8.8)	# CMakePackageConfigHelpers is supported from 2.8.8
+	include(CMakePackageConfigHelpers)
+	write_basic_package_version_file("${PROJECT_BINARY_DIR}/cryptopp-config-version.cmake" VERSION ${cryptopp_VERSION_MAJOR}.${cryptopp_VERSION_MINOR}.${cryptopp_VERSION_PATCH} COMPATIBILITY SameMajorVersion)
+	install(FILES cryptopp-config.cmake ${PROJECT_BINARY_DIR}/cryptopp-config-version.cmake DESTINATION "lib/cmake/cryptopp")
+	install(EXPORT ${export_name} DESTINATION "lib/cmake/cryptopp")
+endif()
 
 # Tests
 if(BUILD_TESTING)

